lib/spack/spack/repo.py

```diff
@@ -852,7 +852,8 @@ def get(self, spec):
 
     def get_pkg_class(self, pkg_name):
         """Find a class for the spec's package and return the class object."""
-        return self.repo_for_pkg(pkg_name).get_pkg_class(pkg_name)
+        with _switch_global_repo(self):
+            return self.repo_for_pkg(pkg_name).get_pkg_class(pkg_name)
```

Comment:
This is another change that needs to be carefully vetted, and that is mainly needed for unit tests. 

`spack external find` uses processes, and those have a different spawning policy depending on the platform. Before, `spack compiler find` was using threads so we didn't have this issue. Maybe we should find a way to enforce these modifications only in tests. :thinking:

Reply:
This was solved in a better way in https://github.com/spack/spack/pull/45053, so removed from this PR